### PR TITLE
Fix jazzy-doc generation

### PR DESCRIPTION
**Pull request by djones6**
## Description
Replaces the `[ci skip]` text for jazzy-doc automated commits with `[jazzy-doc]`, and skips generating jazzy documentation if the latest commit message contains `[jazzy-doc]`.

I also tidied up some of the informational messages around the detection of jazzy-doc tags and required environment variables,  and generate a more specific message in the case of a PR from a fork (we do not support jazzy generation for PRs from forks).

## Motivation and Context

There is a bug relating to the integration of Travis status and Github PRs (https://github.com/travis-ci/travis-ci/issues/2159), such that if the most recent commit contains `[ci skip]`, the PR gets stuck waiting for Travis to run, rather than taking the previous CI result.  This results in an administrator having to use privilege to merge a PR.

Another result of this change is that the docs commit itself does get CI run against it - so if the docs generation were to somehow check in rogue changes, they would be tested.

The `[jazzy-doc]` commit message detection is required to avoid endlessly generating doc commits.  Unfortunately, we can't use `TRAVIS_COMMIT_MESSAGE` and have to go to git directly to get at it, because for PRs, the commit message is for the merge commit that Travis creates locally when testing a PR.  I've coded the extraction of the commit message from the git command line fairly defensively because we don't want this to break (as it could result in endless commits).

## How Has This Been Tested?

I tested using this branch in this PR:
https://github.com/IBM-Swift/Kitura-Markdown/pull/16

## Checklist:

- [x] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [x] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

File: build-package.sh [hunk 1]
```diff
@@ -183,6 +184,7 @@ fi
 # one build (per commit) produces a documentation commit.
 #
 if [ "$(uname)" == "Darwin" -a "${TRAVIS_PULL_REQUEST}" != "false" -a -n "${JAZZY_ELIGIBLE}" ]; then
+  if [ "${TRAVIS_PULL_REQUEST_SLUG}" == "${TRAVIS_REPO_SLUG}" ]; then
```
Comment (EnriqueL8) [inline build-package.sh:187]: Is this changed related?

**Review by EnriqueL8 — APPROVED** [review]

File: build-package.sh (hunk 1 above)
Comment (djones6) [reply build-package.sh:187]: Ah, thanks for spotting that.  It effectively compares the source repo to the destination repo, and only attempts to generate jazzy-doc if they are the same.

Because the Github credentials are defined on our repo, their values are missing when the Travis build runs against someone else's fork (and this also seems to apply to the PR build too).  Even if this were solved, we may not be permitted to push a docs commit to someone else's fork (depending on whether they selected the _Allow edits from maintainers_ option).

So, for now at least, we're restricted to running jazzy-doc updates from our own branches only.
